### DISCUSSION TITLE
Regression - nextcloud AIO upgrade broken 2FA setup

### DISCUSSION THREAD

**Discussion by nuschpl** [nextcloud/all-in-one#2309] [Questions] [answered]
I'm currently running AIO 4.7.0. When initially installing nextcloud configured 2FA for admin user using Google Authenticatior and configured device authentication on some devices. Then had a lot of Nextcloud AIO upgraded, never changed anything in containers manually. Recently got to the state when Nextcloud complains that 2FA was setup for **admin** account but it's unable to load at least one of configured 2FA methods. 
The OCC says it's enabled, so I'm not sure if Nextcloud lost my 2FA config during upgrade ?
```
ncadmin@nextcloud:~$ ./occ.sh twofactorauth:state admin
Two-factor authentication is enabled for user admin

Enabled providers:
- totp
Disabled providers:
- backup_codes
```

**Comment by szaimen** [comment]
Hi, can yo post the output of `sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ config:list twofactor_totp && sudo docker exec --user www-data -it nextcloud-aio-nextcloud ls custom_apps | grep two` ?

**Comment by nuschpl** [comment]
The current output is:
```ncadmin@nextcloud:~$ ./occ.sh config:list twofactor_totp
{
    "apps": {
        "twofactor_totp": {
            "types": "",
            "installed_version": "7.0.0",
            "enabled": "yes"
        }
    }
}
ncadmin@nextcloud:~$ ^C
ncadmin@nextcloud:~$ sudo docker exec --user www-data -it nextcloud-aio-nextcloud ls custom_apps | grep two
contacts	    integration_mastodon  richdocuments  twofactor_totp

```
But unfortunately I've already fixed this - it needed to enable TOTP plugin again. It was disabled or not upgraded. IMHO if there is no guarantee  the plugin always will be updated successuflly then it should disable 2FA

**Comment by szaimen** [comment, marked as answer]
First of all, I could not reproduce this issue. For me the totp app did not get disabled at all during any upsate that I did on my test instance.

---

> IMHO if there is no guarantee the plugin always will be updated successuflly then it should disable 2FA

This is rather something that would need to get implemented in Nextcloud server then.
